packages/rolldown/src/plugin/bindingify-build-hooks.ts

```diff
@@ -119,6 +124,24 @@ export function bindingifyLoad(
       return { code: ret }
     }
 
+    if (!ret.map) {
+      return { code: ret.code }
+    }
+
+    let map =
+      typeof ret.map === 'object'
+        ? ret.map
+        : (JSON.parse(ret.map) as SourceMapInputObject)
+    if (!isEmptySourcemapFiled(map.sources)) {
+      // normalize original sourcemap sources
+      // Port form https://github.com/rollup/rollup/blob/master/src/utils/collapseSourcemaps.ts#L180-L188.
```

Comment:
Is this a temporary workaround? Is the practical way of doing this in rust's `collapseSourcemaps` function?

Reply:
The rust `collapseSourcemaps` is a stable sourcemap remapping.  Here normalize sources path to support rollup cases, it is relevant to remapping.

Reply:
I saw this is for test https://github.com/rollup/rollup/blob/64870c31062983a741832d8cd2d4bcdf55e4de55/test/sourcemaps/samples/loaders/_config.js.

What happen if we have the same logic in rust plugins? 

I saw the test is to ensure the absolute path become relative. So I guess this is the normalization this PR does. Is my understanding right?

![image](https://github.com/rolldown/rolldown/assets/49502170/ff904a28-3f84-43c9-8e5d-5f8db9e1dd2f)

![image](https://github.com/rolldown/rolldown/assets/49502170/792c46ee-3f05-4734-b16f-6ae7bcf6acb1)

Reply:
Here it is intend to test `sourceRoot` feature. You should see here https://github.com/rollup/rollup/blob/64870c31062983a741832d8cd2d4bcdf55e4de55/test/sourcemaps/samples/loaders/_config.js#L27-L30.

Reply:
> What happen if we have the same logic in rust plugins?

The logic maybe should be put into rust side, but I'm not sure at now. It need to loader `sourcemap` is stable, the plugin should normalize `sourceRoot`.